cli/specifier_handler.rs

```diff
@@ -173,12 +207,14 @@ pub struct FetchHandler {
   disk_cache: DiskCache,
   file_fetcher: SourceFileFetcher,
   permissions: Permissions,
+  permissions_dynamic: Permissions,
 }
 
 impl FetchHandler {
   pub fn new(
     program_state: &Arc<ProgramState>,
     permissions: Permissions,
+    permissions_dynamic: Permissions,
```

Comment:
@kitsonk I'm lost here - why do we need two different sets of permissions passes to `FetchHandler`?

Reply:
Because, the module_graph doesn't always wait to fetch dynamic imports when requested by deno_core.  It actually tries to fetch them ahead of time, but does not error if they are missing.  So in the `FetchHandler` we have to ensure we are passing the right set of permissions to the `fetch_handler` to avoid permission escapes when importing dynamic permissions.

In the old world the program_state talked directly to the file_fetcher and would supply the "right" permissions when supplying a dynamic versus static import.

Having said that out loud though, I realise that `program_state` has the right dynamic permissions on it already though, so I should/could use that.

Does that make any more sense?

Reply:
Yes, it does make more sense I think it directly refers to https://github.com/denoland/deno/pull/7996#discussion_r509233924